src/report/ReportPoll.ts

```diff
@@ -0,0 +1,67 @@
+import { MatrixClient } from "matrix-bot-sdk";
+import { ReportManager } from './ReportManager';
+
+class InvalidStateError extends Error {}
+
+export class ReportPoll {
+    private _client: MatrixClient;
+    private _manager: ReportManager;
+    private _save: (a: number) => Promise<any>;
+    private _from = 0;
```

Comment:
are these idiomatic names for private members in TS? (Genuine question because we I don't do that anywhere in Mjolnir :flushed:)